pygsti/forwardsims/forwardsim.py

```diff
@@ -886,10 +886,40 @@ def _bulk_fill_probs_block(self, array_to_fill, layout):
                                                                    outcomes, layout.resource_alloc(), cache, time=None)
 
     def _bulk_fill_dprobs_block(self, array_to_fill, dest_param_slice, layout, param_slice):
-        for element_indices, circuit, outcomes, cache in layout.iter_unique_circuits_with_cache():
-            self._compute_circuit_outcome_probability_derivatives_with_cache(
-                array_to_fill[element_indices, dest_param_slice], circuit, outcomes, param_slice,
-                layout.resource_alloc(), cache)
+
```

Comment:
This finite differencing alternative should be placed elsewhere.  The MapForwardSimulator does finite differencing, but maybe this was placed here for an op-less model (in which case it might go in SuccessFailForwardSimulator).  In any case, this is ugly and finite differencing is already implemented elsewhere.  We should either call an existing implementation or decide to move all finite differencing to the ForwardSimulator base class, which I don't expect we'll want to do.

Reply:
I want to say this hacked in for the opless model, so should probably go to the SuccessFailForwardSimulator. I'll go ahead and make that change.

Reply:
Ok, cool.  I expect the opless model might have finite differencing somewhere in it already -- so maybe check first :)

Reply:
Actually SuccessFailForwardSimulator actually implements _compute_circuit_outcome_probability_derivatives_with_cache(), and down the stack the default is to use finite difference. So maybe my recollection is wrong...

Either way I'm reverting this change and if it breaks something in an RB workflow somewhere, we can find it and fix in the correct spot later.